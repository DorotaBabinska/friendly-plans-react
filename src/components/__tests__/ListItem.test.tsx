import { shallow } from 'enzyme';
import React from 'react';

import { ListItem } from '../ListItem';

describe('ListItem', () => {
  const props = {
    title: 'some title',
    onPress: jest.fn(),
  };

  it('should match snapshot', () => {
    const wrapper = shallow(<ListItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render subtitle if provided', () => {
    const subtitle = 'some additional smaller text';
    const wrapper = shallow(<ListItem {...props} subtitle={subtitle} />);
    expect(wrapper.find(`StyledText[children="${subtitle}"]`)).toExist();
  });

  it('should render icon if provided', () => {
    const icon = { name: 'logout' };
    const wrapper = shallow(<ListItem {...props} icon={icon} />);
    expect(wrapper.find('Icon')).toExist();
  });
});
