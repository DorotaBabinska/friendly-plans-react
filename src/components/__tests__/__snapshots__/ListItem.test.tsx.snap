// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should match snapshot 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  delayPressOut={100}
  onPress={[MockFunction]}
  underlayColor="rgba(0, 0, 0, 0.2)"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingHorizontal": 12,
        "paddingVertical": 16,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "paddingLeft": 12,
        }
      }
    >
      <StyledText
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "Rubik-Regular",
            "fontSize": 15,
            "letterSpacing": 0.15,
            "marginBottom": 2,
          }
        }
      >
        some title
      </StyledText>
    </View>
  </View>
</TouchableHighlight>
`;
